app/validators/form_password_validator.rb

```diff
@@ -40,11 +32,19 @@ def password_graphemes_length
     end
   end
 
-  def password_score
-    @password_score = ZXCVBN_TESTER.test(
-      password,
-      user.email_addresses.flat_map { |address| ForbiddenPasswords.new(address.email).call },
-    )
+  def password_email
+    @emails_strings = user.email_addresses.
+      flat_map { |address| ForbiddenPasswords.new(address.email).call }
+    @emails = false
+    @emails_strings.each do |string|
+      if password.include? string
+        @emails = true
+        break
+      end
+    end
```

Comment:
A few stylistic remarks:

- We won't need the `@` instance assignments unless those variables are referenced in other places, and it appears they're not
- A method like [`any?`](https://ruby-doc.org/core-3.1.1/Enumerable.html#method-i-any-3F) could simplify what you're doing with the separate variable and loop (similar to `Array#some` in JavaScript)
- There might be a deeper conversation to be had about consistency of validation methods overall in our codebase, but at least for consistency within this file, I might expect some sort of negation indicating that the password should _not_ be composed from parts of the user's email.
- It might benefit us to split out some additional methods to make the individual parts a little more readable

Example:

```suggestion
  def password_not_forbidden
    return if !forbidden_passwords.include?(password)
    errors.add(:password, :avoid_using_phrases_that_are_easily_guessed, type: :password)
  end

  def forbidden_passwords
    user.email_addresses.flat_map { |address| ForbiddenPasswords.new(address.email).call }
  end
```

Reply:
To my prior comment, I don't think we're expecting partial matches to be flagged based on current logic.

Reply:
I see. I have that now as you have suggested.